utils/generate/layer.ts

```diff
@@ -27,13 +27,26 @@ export function generateRandomLayer(
       }
 
       const occurrences = layer.assets.map((asset) => asset.occurrence)
-      const rndIndex = weightedRandom(occurrences, layer.assets)
-      assetAttributes.push({
-        layerId: layer.layerId,
-        layerName: layer.layerName,
-        layerOccurrence: layer.layerOccurrence,
-        asset: layer.assets[rndIndex]
-      })
+      if (layer.assets.length > 1) {
+        // requires checking since weightedRandom does not account for assets length of 1
+        const rndIndex = weightedRandom(occurrences, layer.assets)
```

Comment:
I notice you have cases where error can be thrown in `weightedRandom`.
Do you need to add this into a trycatch?

Reply:
will create alert later